frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -322,11 +322,15 @@ export function getXInterval({ settings, series }, xValues, warn) {
 }
 
 export function xValueForWaterfallTotal({ settings, series }) {
+  const xValues = getXValues({ settings, series });
+  const xInterval = getXInterval({ settings, series }, xValues);
+
   if (isTimeseries(settings)) {
-    const xValues = getXValues({ settings, series });
-    const { count, interval } = getXInterval({ settings, series }, xValues);
+    const { count, interval } = xInterval;
     const lastXValue = xValues[xValues.length - 1];
     return lastXValue.clone().add(count, interval);
+  } else if (typeof xInterval === "number") {
```

Comment:
This check feels whacky. @paulrosenzweig is there a better alternative?

Reply:
Yup! I think `isQuantitative` is what you want.

Reply:
I see `isQuantitative(settings) || isHistogram(settings)` is used in `getXInterval`, so it's worth checking if there's an edge case for histograms. I'm not sure what a non-quantitative histogram would be...

Reply:
The histogram looks funny but it definitely works. Meanwhile, I'll adopt your proposed checks.

![image](https://user-images.githubusercontent.com/7288/101521410-ab508780-393a-11eb-95b7-27c86003d567.png)